### PR TITLE
Remove Indirect Settings from interface menu

### LINKED ISSUE #36413: Remove Indirect Settings GUI from Interfaces menu
**Description**
We want to remove the Indirect Settings GUI from the interfaces menu. This should be treated more like a dialog or widget which isn't a `UserSubWindow` because it doesn't provide scientific value.

This will mean we no longer want it to be a `UserSubWindow`, just an ordinary QWidget. The files for this widget should be put in a 'common' location because its used across all the inelastic/indirect interfaces.

One potential problem: how do we connect this GUI to all other open indirect/inelastic interfaces. 

Another thing to consider: is this interface used/useful or should it be removed? Or could it contain more useful/common options?

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR removes the Indirect Settings interface from the Indirect category in the interface menu. This has been removed because it does not add scientific value to the indirect category, however it is still accessible from the cog button at the bottom of each indirect/inelastic interface. It makes more sense for this to be treated as a QWidget rather than an interface in its own right.

Fixes #36413

### To test:
Open any indirect interface
Click on the Settings icon (bottom left)
Adjust a setting and apply
Close and open the settings GUI to check the change was saved.
Open a second Indirect Interface and check that changing a setting will update the setting on all opened Indirect interfaces. The help menu should give a good explanation for what each option does.

The "Settings" interface should not be accessible from the Indirect interface category

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectSettings.cpp [hunk 1]
```diff
@@ -8,16 +8,28 @@
...
 namespace MantidQt::CustomInterfaces {
-DECLARE_SUBWINDOW(IndirectSettings)
 
-IndirectSettings::IndirectSettings(QWidget *parent) : MantidQt::API::UserSubWindow(parent) { m_uiForm.setupUi(this); }
+IndirectSettings::IndirectSettings(QWidget *parent) : QWidget(parent) {
+  auto model = std::make_unique<IndirectSettingsModel>();
+  m_presenter = std::make_unique<IndirectSettingsPresenter>(std::move(model), new IndirectSettingsView(this));
+  m_presenter->subscribeParent(this);
+
+  auto layout = new QGridLayout();
```
Comment (cailafinn) [inline qt/scientific_interfaces/Indirect/IndirectSettings.cpp:22]: Just to confirm, given the the use of `new` here, does Qt handle the garbage collection for these pointers once they've been closed?

**Review by cailafinn — COMMENTED** [review]
Looks good, just wanted to quickly check on the use of `new` here:

File: qt/scientific_interfaces/Indirect/IndirectSettings.cpp (hunk 1 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Indirect/IndirectSettings.cpp:22]: Good question, just had a read of the docs here and it says that the QWidget takes ownership of the QLayout
https://doc.qt.io/qt-5/qwidget.html#setLayout

**Review by cailafinn — APPROVED** [review]
Looks good, works well, much neater. Nice work.

**Pull request merged**